metrics.go

```diff
@@ -67,15 +60,10 @@ func (t *counter) Emit() {
 		return
 	}
 
-	go func(key string, count int) {
-		select {
-		case statQueue <- NewCounter(
-			fmt.Sprintf("%s.%s", metricPrefix, key),
-			count,
-		):
-		default:
-		}
-	}(t.key, t.count)
+	statQueue <- NewCounter(
```

Comment:
AFAIK this is thread safe. The flush happens in a separate go routine so deadlock wouldn't be an issue. I haven't seen any issues when testing. I don't see how race conditions would occur either.

Maybe @sjhitchner can comment. He knows this code better than I do.